File: backend/appointments/src/main/java/com/medical/homevisits/appointments/doctor/entity/Doctor.java
package com.medical.homevisits.appointments.doctor.entity;

import com.medical.homevisits.appointments.workplace.entity.Workplace;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.SuperBuilder;

import java.util.UUID;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@SuperBuilder
@Entity
@Table(name="doctors")
public class Doctor {
    @Id
    private UUID ID;
    private String firstName;
    private String lastName;
    private String specialization;

    @ManyToOne
    @JoinColumn(name="workPlace", nullable = true)
    private Workplace workPlace;
}

File: frontend/app/src/main/java/com/medical/wizytydomowe/api/users/Doctor.kt
package com.medical.wizytydomowe.api.users

import com.medical.wizytydomowe.api.workplace.Workplace
import java.io.Serializable

class Doctor(
    id: String?,
    firstName: String?,
    lastName: String?,
    val specialization: String?,
    val workPlace: Workplace?
) : Serializable, User(id, firstName, lastName)
